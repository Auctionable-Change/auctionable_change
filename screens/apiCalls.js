const url = 'https://auctionable-change-api.herokuapp.com/items'

export const fetchItems = async () => {
  try {
    const response = await fetch(url);
    if(!response.ok) {
      throw new Error(error);
    }
    const data = await response.json();
    return data;
  } catch(error) {
    return false
  }
}

export const postItem = async (listing) => {
  const response = await fetch(url, {
    method: 'POST', 
    headers: {
      'Content-Type': 'application/json'
    },
    body: JSON.stringify(listing)
  })
  return response.json()
}


export const fetchCharities = async (search) => {
  try {
    const response = await fetch(`https://auctionable-change-api.herokuapp.com/charities/${search}`)
    if(!response.ok) {
      throw new Error(error);
    }
    const data = await response.json();
    return data;
  } catch(error) {
    return false
  }
}

let CLOUDINARY_URL = 'https://api.cloudinary.com/v1_1/djk5anakm/upload';


export const cloudinaryPost = (pickerResult, sendEmail) => {
  let photoData;
    let base64Img = `data:image/jpg;base64,${pickerResult.base64}`;
    let data = {
      file: base64Img,
      upload_preset: "oclsdjxp",
    };

    fetch(CLOUDINARY_URL, {
      body: JSON.stringify(data),
      headers: {
        "content-type": "application/json",
      },
      method: "POST",
    })
      .then(async (res) => await res.json())
      .then((res) => (photoData = res.url))
      .then(() => console.log("photo data", photoData))
      .then(() => sendEmail(photoData));
  };

