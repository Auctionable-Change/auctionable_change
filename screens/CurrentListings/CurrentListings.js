import React, { useState, useEffect } from "react";
import {
  View,
  Text,
  Image,
  StyleSheet,
  FlatList,
  ActivityIndicator,
} from "react-native";
import { SafeAreaView } from "react-native-safe-area-context";
import { useStore } from "../../store";
import { fetchItems } from "../apiCalls";
import { Picker, Icon, CardItem, Card, Left, Body, Button } from "native-base";
import NavBar from "../../components/NavBar/NavBar";

const CurrentListings = ({ navigation }) => {
  const { dispatch } = useStore();
  const [listings, setListings] = useState([]);
  const [filterCategory, setFilterCategory] = useState("");
  const [allListings, setAllListings] = useState([]);
  const [isLoading, setIsLoading] = useState(true);

  const filterListings = (filterCriteria) => {
    if (filterCriteria === "all") {
      setListings(allListings);
      setFilterCategory(filterCriteria);
    } else {
      const filteredListings = allListings.filter(
        (listing) => listing.category === filterCriteria
      );
      setListings(filteredListings);
      setFilterCategory(filterCriteria);
    }
  };

  const pressHandler = (name) => {
    let currentListing = listings.filter(
      (listing) => listing.title === name
    )[0];
    dispatch({ type: "ADD_CURRENT_LISTING", currentListing: currentListing });
    navigation.navigate("Details");
  };

  useEffect(() => {
    const fetchData = async () => {
      setIsLoading(true);
      const items = await fetchItems();
      setAllListings(items.items);
      setListings(items.items);
      setIsLoading(false);
    };
    fetchData();
  }, []);

  return (
    <SafeAreaView style={styles.container}>
      <Picker
        mode="dropdown"
        iosIcon={<Icon name="arrow-down" />}
        style={{ alignSelf: "center", width: 300 }}
        placeholder="Filter by Category"
        selectedValue={filterCategory}
        onValueChange={(event) => filterListings(event)}
        textStyle={{ fontFamily: "quicksand-bold", fontSize: 15 }}
      >
        <Picker.Item label="All" value="all" />
        <Picker.Item label="Apparel" value="apparel" />
        <Picker.Item label="Baby/Kids" value="baby" />
        <Picker.Item label="Electronics" value="electronics" />
        <Picker.Item label="Furniture" value="furniture" />
        <Picker.Item label="Home" value="home" />
        <Picker.Item label="Other" value="other" />
      </Picker>

      {isLoading && (
        <View style={{ flex: 1, justifyContent: "center" }}>
          <ActivityIndicator size="large" color="#2CB833" />
        </View>
      )}

      <FlatList
        data={listings}
        keyExtractor={(item) => item.id.toString()}
        renderItem={({ item }) => (
          <Card style={{ marginBottom: 15 }}>
            <CardItem style={styles.cardItem}>
              <Left>
                <Body>
                  <Text style={styles.cardTitle}>{item.title}</Text>
                </Body>
              </Left>
            </CardItem>
            <CardItem cardBody>
              <Image
                source={
                  item.image
                    ? { uri: item.image }
                    : require("../../assets/icons/no-photo-selected.png")
                }
                style={styles.image}
              />
            </CardItem>
            <CardItem style={{ justifyContent: "space-between" }}>
              <Text
                style={{ fontSize: 15, fontFamily: "quicksand" }}
              >{`Minimum Donation: $${item.price}`}</Text>
              <Button
                transparent
                title="Listing Details"
                onPress={() => pressHandler(item.title)}
              >
                <Text style={styles.button}>Listing Details</Text>
              </Button>
            </CardItem>
          </Card>
        )}
      />
      <NavBar navigation={navigation} />
    </SafeAreaView>
  );
};

const styles = StyleSheet.create({
  container: {
    flex: 1,
    flexDirection: "column",
    alignItems: "center",
    justifyContent: "space-between",
    backgroundColor: "#FFFFFF",
  },
  image: {
    width: "100%",
    height: 400,
    resizeMode: "cover",
    alignSelf: "center",
  },
  cardTitle: {
    margin: 5,
    fontSize: 25,
    alignSelf: "center",
    fontFamily: "quicksand-bold",
    textAlign: "center",
  },
  cardItem: {
    shadowOffset: {
      width: 0,
      height: 1,
    },
    shadowOpacity: 0.18,
    shadowRadius: 1.0,
    borderRadius: 3,
  },
  button: {
    color: "#065EFE",
    fontSize: 15,
    fontWeight: "bold",
    fontFamily: "quicksand-bold",
  },
});

export default CurrentListings;
