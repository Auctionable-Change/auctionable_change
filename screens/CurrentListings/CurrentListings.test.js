import React from 'react';
import { render, fireEvent } from 'react-native-testing-library';
import CurrentListings from './CurrentListings';
import { StoreProvider } from '../../store'
import {expect, it} from '@jest/globals';
import renderer from "react-test-renderer";
import { fetchItems } from '../apiCalls'

describe('CurrentListings', () => {
  it('should render', () => {
    const tree = renderer.create(
      <StoreProvider>
        <CurrentListings/>
      </StoreProvider>).toJSON();
    expect(tree).toMatchSnapshot();
  })

  it('should filter listings by category', () => {

    const listings =  [
      {
          "id": 16,
          "title": "Melting iced coffee",
          "description": "Hurry before it’s watered down",
          "price": 1.0,
          "donor": "Lauren",
          "donor_email": "Foxymeatball@aol.com",
          "status": "available",
          "category": "other",
          "charity": "Boys & Girls Clubs of Central Alabama",
          "charity_url": "https://www.charitynavigator.org/?bay=search.summary&orgid=13339&utm_source=DataAPI&utm_content=6bc9f3c8",
          "charity_score": 4,
          "image": "http://res.cloudinary.com/djk5anakm/image/upload/v1595857701/pwc5fwragfvoqeow2ryw.jpg",
          "bids": []
      },
      {
          "id": 17,
          "title": "Melting iced coffee",
          "description": "Hurry before it’s watered down",
          "price": 1.0,
          "donor": "Lauren",
          "donor_email": "Foxymeatball@aol.com",
          "status": "available",
          "category": "other",
          "charity": "Boys & Girls Clubs of Central Alabama",
          "charity_url": "https://www.charitynavigator.org/?bay=search.summary&orgid=13339&utm_source=DataAPI&utm_content=6bc9f3c8",
          "charity_score": 4,
          "image": "http://res.cloudinary.com/djk5anakm/image/upload/v1595857701/pwc5fwragfvoqeow2ryw.jpg",
          "bids": []
      },
      {
          "id": 18,
          "title": "bowl and spoon",
          "description": "Great for cereal ",
          "price": 1.0,
          "donor": "Lauren",
          "donor_email": "Foxymeatball@aol.com",
          "status": "available",
          "category": "home",
          "charity": "Boys & Girls Clubs of Central Alabama",
          "charity_url": "https://www.charitynavigator.org/?bay=search.summary&orgid=13339&utm_source=DataAPI&utm_content=6bc9f3c8",
          "charity_score": 4,
          "image": "http://res.cloudinary.com/djk5anakm/image/upload/v1595862005/ifdtga4y0l3n17d5libd.jpg",
          "bids": []
      }
    ]
    const { getByText } = render(
      <StoreProvider>
        <CurrentListings />
      </StoreProvider>
    )
    
  })
})