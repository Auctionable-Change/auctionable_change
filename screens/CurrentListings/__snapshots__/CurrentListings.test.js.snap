// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CurrentListings should render 1`] = `
<RNCSafeAreaView
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#FFFFFF",
      "flex": 1,
      "flexDirection": "column",
      "justifyContent": "space-between",
    }
  }
>
  <View>
    <View
      accessible={true}
      focusable={true}
      forwardedRef={[Function]}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "center",
          "alignSelf": "center",
          "backgroundColor": "transparent",
          "borderBottomWidth": null,
          "borderColor": "#007aff",
          "borderLeftWidth": null,
          "borderRadius": 5,
          "borderRightWidth": null,
          "borderTopWidth": null,
          "borderWidth": undefined,
          "elevation": 0,
          "flexDirection": "row",
          "height": 45,
          "justifyContent": "space-between",
          "opacity": 1,
          "paddingBottom": 6,
          "paddingTop": 6,
          "shadowColor": null,
          "shadowOffset": null,
          "shadowOpacity": null,
          "shadowRadius": null,
          "width": 300,
        }
      }
    >
      <Text
        ellipsizeMode="tail"
        note={true}
        numberOfLines={1}
        style={
          Array [
            Object {
              "backgroundColor": "transparent",
              "color": "#a7a7a7",
              "fontFamily": "System",
              "fontSize": 16,
              "lineHeight": null,
              "marginLeft": 0,
              "marginRight": 0,
              "paddingLeft": 16,
              "paddingRight": 16,
            },
            undefined,
            undefined,
          ]
        }
      >
        Filter by Category
      </Text>
      <Text />
    </View>
    <Modal
      animationType="slide"
      hardwareAccelerated={false}
      onRequestClose={[Function]}
      supportedOrientations={
        Array [
          "portrait",
          "landscape",
        ]
      }
      transparent={false}
      visible={false}
    >
      <View
        style={
          Object {
            "backgroundColor": "#fff",
            "flex": 1,
            "height": 1334,
          }
        }
      >
        <View>
          <RCTSafeAreaView
            emulateUnlessSupported={true}
            style={
              Object {
                "backgroundColor": "#F8F8F8",
              }
            }
          >
            <View
              style={
                Object {
                  "backgroundColor": "#F8F8F8",
                  "borderBottomColor": "#a7a6ab",
                  "borderBottomWidth": 0.5,
                  "elevation": 3,
                  "flexDirection": "row",
                  "height": 64,
                  "justifyContent": "center",
                  "left": 0,
                  "paddingLeft": 6,
                  "paddingRight": 10,
                  "right": 0,
                  "shadowColor": undefined,
                  "shadowOffset": undefined,
                  "shadowOpacity": undefined,
                  "shadowRadius": undefined,
                  "top": 0,
                }
              }
            >
              <View
                style={
                  Object {
                    "alignItems": "flex-start",
                    "alignSelf": "center",
                    "flex": 1,
                  }
                }
              >
                <View
                  accessible={true}
                  focusable={true}
                  forwardedRef={[Function]}
                  onClick={[Function]}
                  onResponderGrant={[Function]}
                  onResponderMove={[Function]}
                  onResponderRelease={[Function]}
                  onResponderTerminate={[Function]}
                  onResponderTerminationRequest={[Function]}
                  onStartShouldSetResponder={[Function]}
                  style={
                    Object {
                      "alignItems": "center",
                      "alignSelf": null,
                      "backgroundColor": "transparent",
                      "borderBottomWidth": null,
                      "borderColor": null,
                      "borderLeftWidth": null,
                      "borderRadius": 5,
                      "borderRightWidth": null,
                      "borderTopWidth": null,
                      "borderWidth": undefined,
                      "elevation": 0,
                      "flexDirection": "row",
                      "height": 45,
                      "justifyContent": "space-between",
                      "marginLeft": 3,
                      "opacity": 1,
                      "paddingBottom": 6,
                      "paddingLeft": 4,
                      "paddingRight": 6,
                      "paddingTop": 6,
                      "shadowColor": null,
                      "shadowOffset": null,
                      "shadowOpacity": null,
                      "shadowRadius": null,
                    }
                  }
                >
                  <Text
                    style={
                      Object {
                        "backgroundColor": "transparent",
                        "color": "#007aff",
                        "fontFamily": "System",
                        "fontSize": 17,
                        "marginLeft": 0,
                        "marginRight": 0,
                        "paddingLeft": 2,
                        "paddingRight": undefined,
                        "top": 1,
                      }
                    }
                  >
                    Back
                  </Text>
                </View>
              </View>
              <View
                style={
                  Object {
                    "alignItems": "center",
                    "alignSelf": "center",
                    "flex": 1,
                  }
                }
              >
                <Text
                  numberOfLines={1}
                  style={
                    Object {
                      "color": "#000",
                      "fontFamily": "System",
                      "fontSize": 17,
                      "fontWeight": "700",
                      "marginLeft": undefined,
                      "paddingLeft": 4,
                      "paddingTop": 1,
                      "textAlign": "center",
                    }
                  }
                >
                  Select One
                </Text>
              </View>
              <View
                style={
                  Object {
                    "alignItems": "flex-end",
                    "alignSelf": "center",
                    "flex": 1,
                    "flexDirection": "row",
                    "justifyContent": "flex-end",
                  }
                }
              />
            </View>
          </RCTSafeAreaView>
        </View>
        <RCTSafeAreaView
          emulateUnlessSupported={true}
          style={
            Object {
              "backgroundColor": "transparent",
              "flex": 1,
            }
          }
        >
          <RCTScrollView
            automaticallyAdjustContentInsets={false}
            contentContainerStyle={
              Array [
                Object {
                  "padding": undefined,
                },
                undefined,
              ]
            }
            contentInset={
              Object {
                "bottom": 0,
              }
            }
            enableAutomaticScroll={true}
            enableOnAndroid={false}
            enableResetScrollToCoords={true}
            extraHeight={75}
            extraScrollHeight={0}
            getScrollResponder={[Function]}
            handleOnScroll={[Function]}
            keyboardDismissMode="interactive"
            keyboardOpeningTime={250}
            keyboardShouldPersistTaps="handled"
            keyboardSpace={0}
            onScroll={[Function]}
            resetKeyboardSpace={[Function]}
            resetScrollToCoords={
              Object {
                "x": 0,
                "y": 0,
              }
            }
            scrollEventThrottle={1}
            scrollForExtraHeightOnAndroid={[Function]}
            scrollIntoView={[Function]}
            scrollToEnd={[Function]}
            scrollToFocusedInput={[Function]}
            scrollToPosition={[Function]}
            showsVerticalScrollIndicator={true}
            style={
              Object {
                "backgroundColor": "transparent",
                "flex": 1,
              }
            }
            update={[Function]}
            viewIsInsideTabBar={false}
          >
            <View>
              <RCTScrollView
                data={
                  Array [
                    <Item
                      label="All"
                      value="all"
                    />,
                    <Item
                      label="Electronics"
                      value="electronics"
                    />,
                    <Item
                      label="Home"
                      value="home"
                    />,
                    <Item
                      label="Furniture"
                      value="furniture"
                    />,
                    <Item
                      label="Baby/Kids"
                      value="baby"
                    />,
                    <Item
                      label="Other"
                      value="other"
                    />,
                  ]
                }
                disableVirtualization={false}
                getItem={[Function]}
                getItemCount={[Function]}
                horizontal={false}
                initialNumToRender={10}
                keyExtractor={[Function]}
                maxToRenderPerBatch={10}
                onContentSizeChange={[Function]}
                onEndReachedThreshold={2}
                onLayout={[Function]}
                onMomentumScrollEnd={[Function]}
                onScroll={[Function]}
                onScrollBeginDrag={[Function]}
                onScrollEndDrag={[Function]}
                removeClippedSubviews={false}
                renderItem={[Function]}
                scrollEventThrottle={50}
                stickyHeaderIndices={Array []}
                updateCellsBatchingPeriod={50}
                viewabilityConfigCallbackPairs={Array []}
                windowSize={21}
              >
                <View>
                  <View
                    onLayout={[Function]}
                    style={null}
                  >
                    <View
                      accessible={true}
                      focusable={true}
                      onClick={[Function]}
                      onResponderGrant={[Function]}
                      onResponderMove={[Function]}
                      onResponderRelease={[Function]}
                      onResponderTerminate={[Function]}
                      onResponderTerminationRequest={[Function]}
                      onStartShouldSetResponder={[Function]}
                    >
                      <View
                        button={true}
                        onPress={[Function]}
                        selected={false}
                        style={
                          Object {
                            "alignItems": "center",
                            "backgroundColor": "transparent",
                            "borderBottomWidth": 0.5,
                            "borderColor": "#c9c9c9",
                            "flexDirection": "row",
                            "marginLeft": 16,
                            "paddingBottom": 13,
                            "paddingRight": 16,
                            "paddingTop": 13,
                          }
                        }
                      >
                        <View
                          style={
                            Object {
                              "alignItems": "flex-start",
                              "alignSelf": "center",
                              "flex": 1,
                              "flexDirection": "row",
                            }
                          }
                        >
                          <Text
                            style={
                              Object {
                                "alignSelf": "center",
                                "color": "#000",
                                "fontFamily": "System",
                                "fontSize": 16,
                              }
                            }
                          >
                            All
                          </Text>
                        </View>
                        <View
                          style={
                            Object {
                              "alignItems": "flex-end",
                              "alignSelf": "center",
                              "flex": 0.28,
                              "paddingBottom": null,
                              "paddingLeft": null,
                              "paddingRight": null,
                              "paddingTop": null,
                            }
                          }
                        >
                          <View
                            accessible={true}
                            focusable={false}
                            forwardedRef={[Function]}
                            onClick={[Function]}
                            onResponderGrant={[Function]}
                            onResponderMove={[Function]}
                            onResponderRelease={[Function]}
                            onResponderTerminate={[Function]}
                            onResponderTerminationRequest={[Function]}
                            onStartShouldSetResponder={[Function]}
                            style={
                              Object {
                                "alignSelf": null,
                                "opacity": 1,
                              }
                            }
                          />
                        </View>
                      </View>
                    </View>
                  </View>
                  <View
                    onLayout={[Function]}
                    style={null}
                  >
                    <View
                      accessible={true}
                      focusable={true}
                      onClick={[Function]}
                      onResponderGrant={[Function]}
                      onResponderMove={[Function]}
                      onResponderRelease={[Function]}
                      onResponderTerminate={[Function]}
                      onResponderTerminationRequest={[Function]}
                      onStartShouldSetResponder={[Function]}
                    >
                      <View
                        button={true}
                        onPress={[Function]}
                        selected={false}
                        style={
                          Object {
                            "alignItems": "center",
                            "backgroundColor": "transparent",
                            "borderBottomWidth": 0.5,
                            "borderColor": "#c9c9c9",
                            "flexDirection": "row",
                            "marginLeft": 16,
                            "paddingBottom": 13,
                            "paddingRight": 16,
                            "paddingTop": 13,
                          }
                        }
                      >
                        <View
                          style={
                            Object {
                              "alignItems": "flex-start",
                              "alignSelf": "center",
                              "flex": 1,
                              "flexDirection": "row",
                            }
                          }
                        >
                          <Text
                            style={
                              Object {
                                "alignSelf": "center",
                                "color": "#000",
                                "fontFamily": "System",
                                "fontSize": 16,
                              }
                            }
                          >
                            Electronics
                          </Text>
                        </View>
                        <View
                          style={
                            Object {
                              "alignItems": "flex-end",
                              "alignSelf": "center",
                              "flex": 0.28,
                              "paddingBottom": null,
                              "paddingLeft": null,
                              "paddingRight": null,
                              "paddingTop": null,
                            }
                          }
                        >
                          <View
                            accessible={true}
                            focusable={false}
                            forwardedRef={[Function]}
                            onClick={[Function]}
                            onResponderGrant={[Function]}
                            onResponderMove={[Function]}
                            onResponderRelease={[Function]}
                            onResponderTerminate={[Function]}
                            onResponderTerminationRequest={[Function]}
                            onStartShouldSetResponder={[Function]}
                            style={
                              Object {
                                "alignSelf": null,
                                "opacity": 1,
                              }
                            }
                          />
                        </View>
                      </View>
                    </View>
                  </View>
                  <View
                    onLayout={[Function]}
                    style={null}
                  >
                    <View
                      accessible={true}
                      focusable={true}
                      onClick={[Function]}
                      onResponderGrant={[Function]}
                      onResponderMove={[Function]}
                      onResponderRelease={[Function]}
                      onResponderTerminate={[Function]}
                      onResponderTerminationRequest={[Function]}
                      onStartShouldSetResponder={[Function]}
                    >
                      <View
                        button={true}
                        onPress={[Function]}
                        selected={false}
                        style={
                          Object {
                            "alignItems": "center",
                            "backgroundColor": "transparent",
                            "borderBottomWidth": 0.5,
                            "borderColor": "#c9c9c9",
                            "flexDirection": "row",
                            "marginLeft": 16,
                            "paddingBottom": 13,
                            "paddingRight": 16,
                            "paddingTop": 13,
                          }
                        }
                      >
                        <View
                          style={
                            Object {
                              "alignItems": "flex-start",
                              "alignSelf": "center",
                              "flex": 1,
                              "flexDirection": "row",
                            }
                          }
                        >
                          <Text
                            style={
                              Object {
                                "alignSelf": "center",
                                "color": "#000",
                                "fontFamily": "System",
                                "fontSize": 16,
                              }
                            }
                          >
                            Home
                          </Text>
                        </View>
                        <View
                          style={
                            Object {
                              "alignItems": "flex-end",
                              "alignSelf": "center",
                              "flex": 0.28,
                              "paddingBottom": null,
                              "paddingLeft": null,
                              "paddingRight": null,
                              "paddingTop": null,
                            }
                          }
                        >
                          <View
                            accessible={true}
                            focusable={false}
                            forwardedRef={[Function]}
                            onClick={[Function]}
                            onResponderGrant={[Function]}
                            onResponderMove={[Function]}
                            onResponderRelease={[Function]}
                            onResponderTerminate={[Function]}
                            onResponderTerminationRequest={[Function]}
                            onStartShouldSetResponder={[Function]}
                            style={
                              Object {
                                "alignSelf": null,
                                "opacity": 1,
                              }
                            }
                          />
                        </View>
                      </View>
                    </View>
                  </View>
                  <View
                    onLayout={[Function]}
                    style={null}
                  >
                    <View
                      accessible={true}
                      focusable={true}
                      onClick={[Function]}
                      onResponderGrant={[Function]}
                      onResponderMove={[Function]}
                      onResponderRelease={[Function]}
                      onResponderTerminate={[Function]}
                      onResponderTerminationRequest={[Function]}
                      onStartShouldSetResponder={[Function]}
                    >
                      <View
                        button={true}
                        onPress={[Function]}
                        selected={false}
                        style={
                          Object {
                            "alignItems": "center",
                            "backgroundColor": "transparent",
                            "borderBottomWidth": 0.5,
                            "borderColor": "#c9c9c9",
                            "flexDirection": "row",
                            "marginLeft": 16,
                            "paddingBottom": 13,
                            "paddingRight": 16,
                            "paddingTop": 13,
                          }
                        }
                      >
                        <View
                          style={
                            Object {
                              "alignItems": "flex-start",
                              "alignSelf": "center",
                              "flex": 1,
                              "flexDirection": "row",
                            }
                          }
                        >
                          <Text
                            style={
                              Object {
                                "alignSelf": "center",
                                "color": "#000",
                                "fontFamily": "System",
                                "fontSize": 16,
                              }
                            }
                          >
                            Furniture
                          </Text>
                        </View>
                        <View
                          style={
                            Object {
                              "alignItems": "flex-end",
                              "alignSelf": "center",
                              "flex": 0.28,
                              "paddingBottom": null,
                              "paddingLeft": null,
                              "paddingRight": null,
                              "paddingTop": null,
                            }
                          }
                        >
                          <View
                            accessible={true}
                            focusable={false}
                            forwardedRef={[Function]}
                            onClick={[Function]}
                            onResponderGrant={[Function]}
                            onResponderMove={[Function]}
                            onResponderRelease={[Function]}
                            onResponderTerminate={[Function]}
                            onResponderTerminationRequest={[Function]}
                            onStartShouldSetResponder={[Function]}
                            style={
                              Object {
                                "alignSelf": null,
                                "opacity": 1,
                              }
                            }
                          />
                        </View>
                      </View>
                    </View>
                  </View>
                  <View
                    onLayout={[Function]}
                    style={null}
                  >
                    <View
                      accessible={true}
                      focusable={true}
                      onClick={[Function]}
                      onResponderGrant={[Function]}
                      onResponderMove={[Function]}
                      onResponderRelease={[Function]}
                      onResponderTerminate={[Function]}
                      onResponderTerminationRequest={[Function]}
                      onStartShouldSetResponder={[Function]}
                    >
                      <View
                        button={true}
                        onPress={[Function]}
                        selected={false}
                        style={
                          Object {
                            "alignItems": "center",
                            "backgroundColor": "transparent",
                            "borderBottomWidth": 0.5,
                            "borderColor": "#c9c9c9",
                            "flexDirection": "row",
                            "marginLeft": 16,
                            "paddingBottom": 13,
                            "paddingRight": 16,
                            "paddingTop": 13,
                          }
                        }
                      >
                        <View
                          style={
                            Object {
                              "alignItems": "flex-start",
                              "alignSelf": "center",
                              "flex": 1,
                              "flexDirection": "row",
                            }
                          }
                        >
                          <Text
                            style={
                              Object {
                                "alignSelf": "center",
                                "color": "#000",
                                "fontFamily": "System",
                                "fontSize": 16,
                              }
                            }
                          >
                            Baby/Kids
                          </Text>
                        </View>
                        <View
                          style={
                            Object {
                              "alignItems": "flex-end",
                              "alignSelf": "center",
                              "flex": 0.28,
                              "paddingBottom": null,
                              "paddingLeft": null,
                              "paddingRight": null,
                              "paddingTop": null,
                            }
                          }
                        >
                          <View
                            accessible={true}
                            focusable={false}
                            forwardedRef={[Function]}
                            onClick={[Function]}
                            onResponderGrant={[Function]}
                            onResponderMove={[Function]}
                            onResponderRelease={[Function]}
                            onResponderTerminate={[Function]}
                            onResponderTerminationRequest={[Function]}
                            onStartShouldSetResponder={[Function]}
                            style={
                              Object {
                                "alignSelf": null,
                                "opacity": 1,
                              }
                            }
                          />
                        </View>
                      </View>
                    </View>
                  </View>
                  <View
                    onLayout={[Function]}
                    style={null}
                  >
                    <View
                      accessible={true}
                      focusable={true}
                      onClick={[Function]}
                      onResponderGrant={[Function]}
                      onResponderMove={[Function]}
                      onResponderRelease={[Function]}
                      onResponderTerminate={[Function]}
                      onResponderTerminationRequest={[Function]}
                      onStartShouldSetResponder={[Function]}
                    >
                      <View
                        button={true}
                        onPress={[Function]}
                        selected={false}
                        style={
                          Object {
                            "alignItems": "center",
                            "backgroundColor": "transparent",
                            "borderBottomWidth": 0.5,
                            "borderColor": "#c9c9c9",
                            "flexDirection": "row",
                            "marginLeft": 16,
                            "paddingBottom": 13,
                            "paddingRight": 16,
                            "paddingTop": 13,
                          }
                        }
                      >
                        <View
                          style={
                            Object {
                              "alignItems": "flex-start",
                              "alignSelf": "center",
                              "flex": 1,
                              "flexDirection": "row",
                            }
                          }
                        >
                          <Text
                            style={
                              Object {
                                "alignSelf": "center",
                                "color": "#000",
                                "fontFamily": "System",
                                "fontSize": 16,
                              }
                            }
                          >
                            Other
                          </Text>
                        </View>
                        <View
                          style={
                            Object {
                              "alignItems": "flex-end",
                              "alignSelf": "center",
                              "flex": 0.28,
                              "paddingBottom": null,
                              "paddingLeft": null,
                              "paddingRight": null,
                              "paddingTop": null,
                            }
                          }
                        >
                          <View
                            accessible={true}
                            focusable={false}
                            forwardedRef={[Function]}
                            onClick={[Function]}
                            onResponderGrant={[Function]}
                            onResponderMove={[Function]}
                            onResponderRelease={[Function]}
                            onResponderTerminate={[Function]}
                            onResponderTerminationRequest={[Function]}
                            onStartShouldSetResponder={[Function]}
                            style={
                              Object {
                                "alignSelf": null,
                                "opacity": 1,
                              }
                            }
                          />
                        </View>
                      </View>
                    </View>
                  </View>
                </View>
              </RCTScrollView>
            </View>
          </RCTScrollView>
        </RCTSafeAreaView>
      </View>
    </Modal>
  </View>
  <View
    style={
      Object {
        "flex": 1,
        "justifyContent": "center",
      }
    }
  >
    <ActivityIndicator
      animating={true}
      color="#2CB833"
      hidesWhenStopped={true}
      size="large"
    />
  </View>
  <RCTScrollView
    data={Array []}
    disableVirtualization={false}
    getItem={[Function]}
    getItemCount={[Function]}
    horizontal={false}
    initialNumToRender={10}
    keyExtractor={[Function]}
    maxToRenderPerBatch={10}
    onContentSizeChange={[Function]}
    onEndReachedThreshold={2}
    onLayout={[Function]}
    onMomentumScrollEnd={[Function]}
    onScroll={[Function]}
    onScrollBeginDrag={[Function]}
    onScrollEndDrag={[Function]}
    removeClippedSubviews={false}
    renderItem={[Function]}
    scrollEventThrottle={50}
    stickyHeaderIndices={Array []}
    updateCellsBatchingPeriod={50}
    viewabilityConfigCallbackPairs={Array []}
    windowSize={21}
  >
    <View />
  </RCTScrollView>
  <View
    style={
      Object {
        "backgroundColor": "white",
        "borderTopColor": "#dddddd",
        "borderTopWidth": 1,
        "bottom": "0%",
        "display": "flex",
        "flexDirection": "row",
        "height": "8%",
        "justifyContent": "space-around",
        "padding": 8,
        "position": "absolute",
        "width": "100%",
      }
    }
  >
    <RNGestureHandlerButton
      collapsable={false}
      onGestureEvent={[Function]}
      onGestureHandlerEvent={[Function]}
      onGestureHandlerStateChange={[Function]}
      onHandlerStateChange={[Function]}
      rippleColor={0}
      style={
        Array [
          Object {
            "overflow": "hidden",
          },
          undefined,
        ]
      }
    >
      <View
        accessible={true}
      >
        <Image
          accessibilityRole="button"
          id="search"
          source={1}
          style={
            Object {
              "height": 30,
              "width": 30,
            }
          }
        />
      </View>
    </RNGestureHandlerButton>
    <RNGestureHandlerButton
      collapsable={false}
      onGestureEvent={[Function]}
      onGestureHandlerEvent={[Function]}
      onGestureHandlerStateChange={[Function]}
      onHandlerStateChange={[Function]}
      rippleColor={0}
      style={
        Array [
          Object {
            "overflow": "hidden",
          },
          undefined,
        ]
      }
    >
      <View
        accessible={true}
      >
        <Image
          accessibilityRole="button"
          id="add"
          source={1}
          style={
            Object {
              "height": 30,
              "width": 30,
            }
          }
        />
      </View>
    </RNGestureHandlerButton>
    <RNGestureHandlerButton
      collapsable={false}
      onGestureEvent={[Function]}
      onGestureHandlerEvent={[Function]}
      onGestureHandlerStateChange={[Function]}
      onHandlerStateChange={[Function]}
      rippleColor={0}
      style={
        Array [
          Object {
            "overflow": "hidden",
          },
          undefined,
        ]
      }
    >
      <View
        accessible={true}
      >
        <Image
          accessibilityRole="button"
          id="home"
          source={1}
          style={
            Object {
              "height": 30,
              "width": 30,
            }
          }
        />
      </View>
    </RNGestureHandlerButton>
  </View>
</RNCSafeAreaView>
`;
